presto-main/src/main/java/com/facebook/presto/operator/scalar/SplitToMapFunction.java

```diff
@@ -77,8 +76,14 @@ public Block splitToMap(@TypeParameter("map<varchar,varchar>") Type mapType, @Sq
                 if (value.indexOf(keyValueDelimiter) >= 0) {
                     throw new PrestoException(INVALID_FUNCTION_ARGUMENT, "Key-value delimiter must appear exactly once in each entry. Bad input: '" + keyValuePair.toStringUtf8() + "'");
                 }
+
                 if (map.containsKey(key)) {
-                    throw new PrestoException(INVALID_FUNCTION_ARGUMENT, format("Duplicate keys (%s) are not allowed", key.toStringUtf8()));
+                    if (function == null) {
+                        throw new PrestoException(INVALID_FUNCTION_ARGUMENT, format("Duplicate keys (%s) are not allowed. Specifying a lambda to resolve conflicts can " +
```

Comment:
Nits: If we want to break this into multiple lines, I personally prefer the following format:
```
throw new PrestoException(
    INVALID_FUNCTION_ARGUMENT,
    format("Duplicate keys (%s) are not allowed. Specifying a lambda to resolve conflicts can avoid this error",  key.toStringUtf8()));
```